specification/flag-evaluation/flag-evaluation.json

```diff
@@ -0,0 +1,140 @@
+        "RFC 2119 keyword": "MUST"
+    },
+    {
+        "id": "1.6",
+        "clean id": "1_6",
+        "content": "The client MUST provide a method to add `hooks` which accepts one or more API-conformant `hooks`, and appends them to the collection of any previously added hooks. When new hooks are added, previously added hooks are not removed.",
+        "RFC 2119 keyword": "MUST"
+    },
+    {
+        "id": "1.7",
+        "clean id": "1_7",
+        "content": "The `client` MUST provide methods for flag evaluation, with parameters `flag key` (string, required), `default value` (boolean | number | string | structure, required), `evaluation context` (optional), and `evaluation options` (optional).",
+        "RFC 2119 keyword": "MUST"
+    },
+    {
+        "id": "1.9",
+        "clean id": "1_9",
+        "content": "The `client` MUST provide methods for detailed flag value evaluation with the value wrapped in a `evaluation details` structure, with parameters `flag key` (string, required), `default value` (boolean | number | string | structure, required), `evaluation context` (optional), and `evaluation options` (optional).",
+        "RFC 2119 keyword": "MUST"
+    },
+    {
+        "id": "1.10",
+        "clean id": "1_10",
+        "content": "The `evaluation details` structure's `value` field MUST contain the evaluated flag value.",
+        "RFC 2119 keyword": "MUST"
+    },
+    {
+        "id": "1.12",
+        "clean id": "1_12",
+        "content": "The `evaluation details` structure's `flag key` field MUST must contain the `flag key` argument passed to the detailed flag evaluation method.",
+        "RFC 2119 keyword": "MUST"
+    },
+    {
+        "id": "1.13",
+        "clean id": "1_13",
+        "content": "The `evaluation details` structure's `variant` field SHOULD contain a identifier logically corresponding to the return flag value.",
+        "RFC 2119 keyword": "SHOULD"
+    },
+    {
+        "id": "1.14",
+        "clean id": "1_14",
+        "content": "The `evaluation details` structure's `reason` field SHOULD indicate the semantic reason for the returned flag value. Possible values vary by provider, but may include `\"TARGETING_MATCH\"`, `\"SPLIT\"`, `\"DISABLED\"`, `\"DEFAULT\"`, `\"UNKNOWN\"` and `\"ERROR\"`.",
+        "RFC 2119 keyword": "SHOULD"
+    },
+    {
+        "id": "1.15",
+        "clean id": "1_15",
+        "content": "In cases of abnormal execution (network failure, unhandled error, etc) the `reason` field in the `evaluation details` SHOULD be set to `\"ERROR\"`.",
+        "RFC 2119 keyword": "SHOULD"
+    },
+    {
+        "id": "1.16",
+        "clean id": "1_16",
+        "content": "In cases of abnormal execution, the `evaluation details` structure's `error code` field SHOULD identify an error occurred during flag evaluation, with possible values `\"PROVIDER_NOT_READY\"`, `\"FLAG_NOT_FOUND\"`, `\"PARSE_ERROR\"`, `\"TYPE_MISMATCH\"`, or `\"GENERAL\"`.",
```

Comment:
are you literally having the structure have a field called `error code`, or is "error code" a generic name for the field that will have a different string literal in the code?

If the former, then I would consider renaming it so that the field name doesn't have whitespace in it. If the latter, then that's fine (albeit slightly confusing to me as it stands)

Reply:
No, this is just an effort to stay language neutral, since there's differing casing-conventions. I certainly don't recommend whitespace in fields. See the style guide (and feel free to disagree with it: https://github.com/open-feature/spec/blob/spec/flag-evaluation-api/README.md#style-guide)

> Use "sentence case" enclosed in ticks (`) when identifying entities outside of code blocks (ie: evaluation details instead of EvaluationDetails).